src/form/fields/AutoField.tsx

```diff
@@ -18,8 +18,28 @@ export const AutoField: FunctionComponent<FieldProps> = ({ propName, required, o
     throw new Error(`AutoField: formComponentFactory is not defined for ${propName}`);
   }
 
-  const isFieldDefined =
-    schema.type === 'object' ? isDefined(value) && Object.keys(value).length > 0 : isDefined(value);
+  let isFieldDefined = false;
+  if (isDefined(value)) {
+    // Check if the value is a valid value
+    switch (schema.type) {
+      case 'object':
+        isFieldDefined = typeof value === 'object' && Object.keys(value).length > 0;
+        break;
+      case 'array':
+        isFieldDefined = Array.isArray(value) && value.length > 0;
+        break;
+      case 'boolean':
+        isFieldDefined = typeof value === 'boolean';
+        break;
+      case 'string':
+      case 'number':
+        isFieldDefined = value !== '';
```

Comment:
What would happen if the value is `45`? Wouldn't this check wrongly assume the number it's not defined?

Reply:
Oh, I see, this is valuable insight. Could you please put this information in a comment in the code? I think without the context you provided, we might be inclined to do a different type of comparison. If you added tests for it, it would also help for sure.